include/modem/lte_lc.h

```diff
@@ -163,6 +163,12 @@ enum lte_lc_evt_type {
 	 *  the TAU happens, thus saving power by avoiding sending data and the TAU separately.
 	 */
 	LTE_LC_EVT_TAU_PRE_WARNING,
+
+	/** Event containing results from neighbor cell measurements. */
+	LTE_LC_EVT_NEIGHBOR_CELLS,
+
+	/** No cells were found during neightbor cell measurements. */
+	LTE_LC_EVT_NEIGHBOR_CELLS_NONE,
```

Comment:
Is there a need for a separate event when no neighbor cells are found? In that case `ncells_count` would be zero.

As a generic comment, there's already a lot of events which have associated data in the union. It would be good to say which enum/struct is used with which event.

Reply:
> Is there a need for a separate event when no neighbor cells are found? In that case ncells_count would be zero.

This only had a purpose at a point in development, but doesn't really anymore, so I'll remove the event type.

> As a generic comment, there's already a lot of events which have associated data in the union. It would be good to say which enum/struct is used with which event.

Agreed. This is planned to come in a separate PR for all events, together with updated readme and general docs overhaul for the library.

Reply:
> > Is there a need for a separate event when no neighbor cells are found? In that case ncells_count would be zero.

> This only had a purpose at a point in development, but doesn't really anymore, so I'll remove the event type.

Ok, good. The name of the event could also be for example `LTE_LC_EVT_NEIGHBOR_CELL_MEAS`. That way it would refer to the neighbor cell measurement operation. The data would then contain zero or more neighbor cells.

Reply:
Updated to the suggested event name, it's better. 
`LTE_LC_EVT_NEIGHBOR_CELL_MEAS` is now also used in the case where measurement status is `1`, meaning no current cell, and in the case where there is a current cell, but no neighbor cell information.  `ncells_count` is set to `0` in both cases and `struct lte_lc_cells_info` documentation is added to clarify that the cell information should only be considerd valid when `ncells_count > 0`.